receiver/azureeventhubreceiver/azure_formatter_impl.go

```diff
@@ -0,0 +1,204 @@
+func asTimestamp(s string) (pcommon.Timestamp, error) {
+	t, err := iso8601.ParseString(s)
+	if err != nil {
+		return 0, err
+	}
+	return pcommon.Timestamp(t.UnixNano()), nil
+}
+
+// asSeverity converts the Azure log level to equivalent
+// OpenTelemetry severity numbers. If the log level is not
+// valid, then the 'Unspecified' value is returned.
+func asSeverity(s string) plog.SeverityNumber {
+	switch s {
+	case "Informational":
+		return plog.SeverityNumberInfo
+	case "Warning":
+		return plog.SeverityNumberWarn
+	case "Error":
+		return plog.SeverityNumberError
+	case "Critical":
+		return plog.SeverityNumberFatal
+	default:
+		return plog.SeverityNumberUnspecified
+	}
+}
+
+// setIf will modify the given raw map by setting
+// the key and value iff the value is not null and
+// not the empty string.
+func setIf(attrs map[string]interface{}, key string, value *string) {
+	if value != nil && *value != "" {
+		attrs[key] = *value
+	}
+}
+
+// extractRawAttributes creates a raw attribute map and
+// inserts attributes from the Azure log record. Optional
+// attributes are only inserted if they are defined. The
+// azureDuration value is only set if the value in the
+// Azure log record can be parsed as an integer.
+func extractRawAttributes(log azureLogRecord) map[string]interface{} {
+	var attrs = map[string]interface{}{}
+
+	attrs[azureCategory] = log.Category
+	setIf(attrs, azureCorrelationID, log.CorrelationID)
+	if log.DurationMs != nil {
+		duration, err := strconv.ParseInt(*log.DurationMs, 10, 64)
+		if err == nil {
+			attrs[azureDuration] = duration
+		}
+	}
+	if log.Identity != nil {
+		attrs[azureIdentity] = *log.Identity
+	}
+	attrs[azureOperationName] = log.OperationName
+	setIf(attrs, azureOperationVersion, log.OperationVersion)
+	if log.Properties != nil {
+		attrs[azureProperties] = *log.Properties
+	}
+	setIf(attrs, azureResultDescription, log.ResultDescription)
+	setIf(attrs, azureResultSignature, log.ResultSignature)
+	setIf(attrs, azureResultType, log.ResultType)
+	setIf(attrs, azureTenantID, log.TenantID)
+
+	setIf(attrs, cloudRegion, log.Location)
+	attrs[cloudProvider] = "azure"
+
+	setIf(attrs, netSockPeerAddr, log.CallerIPAddress)
+
+	return attrs
+}
+
+// transform takes a byte array containing a JSON-encoded
+// payload with Azure log records and transforms it into
+// an OpenTelemetry plog.Logs object. The data in the Azure
+// log record appears as fields and attributes in the
+// OpenTelemetry representation; the bodies of the
+// OpenTelemetry log records are empty.
+func transform(buildInfo component.BuildInfo, data []byte) (*plog.Logs, error) {
+
+	l := plog.NewLogs()
+
+	var azureLogs azureRecords
+	decoder := json.NewDecoder(bytes.NewReader(data))
+	if err := decoder.Decode(&azureLogs); err != nil {
+		return &l, err
+	}
+
+	resourceLogs := l.ResourceLogs().AppendEmpty()
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	scopeLogs.Scope().SetName(fmt.Sprintf("otelcol/%s", typeStr))
```

Comment:
`fmt.Sprintf("otelcol/%s", typeStr)` is static, making this constant would reduce GC efforts.

Reply:
Done.